lib/fork.js

```diff
@@ -18,13 +18,29 @@ module.exports = function (args) {
 
 	var ps = childProcess.fork(babel, args, options);
 
+	var killed = false;
+	var killedPromise = new Promise(function (resolve) {
```

Comment:
We need `killedPromise` in [`cli.js#exit`](https://github.com/jamestalmage/ava/blob/7087a08dda47b5d7549d4b93f90fe22d49816acf/cli.js#L162) to wait for all the child processes to exit before killing the parent.

This chunk of code certainly could live inside the other Promise, but then you have a Promise inside a Promise and need to to do something like the following (which seems less clean to):

```js
var promise = new Promise(function (resolveOuterPromise, rejectOuterPromise) {
  // ...
  var killedPromise = new Promise(function (resolveInnerPromise) {
    ps.on('exit', function (code) {
      resolveInnerPromise(code);
      if (code > 0) {
        rejectOuterPromise(code);
      }     
  });
});
```